src/gallery/store.js

```diff
@@ -97,22 +97,22 @@ export default new Vuex.Store({
 
     buildStandaloneData: function(state, inputArray) {
       state.library = state.library || {};
-      _.each(inputArray, function( item ) { 
-        state.library.[item.key] = Object.freeze( item.value ); 
+      _.each(inputArray, function( item ) {
+        state.library[item.key] = Object.freeze( item.value );
```

Comment:
This is the actual fix.  Sorry, I didn't realise there were so many whitespace changes.

Reply:
I'm pretty sure this a non issue related to how I don't really care much about committing fully functional code because it's all experimental until a released build (I'll explain that)... **or** the issue could also be related to you testing it using the `watch:dev` command. So I think they're both basically causing the same issue. Jszip doesn't like it when you try to serve it files that do not exist and this causes the saving process to stop. 

Now there are 3 chunk files that I've not been able to give a name to and their `contenthash` in the filename keeps changing every now and then so probably in quite a few commits I've not updated the `save-gallery.vue` → `files` array and it's trying to add files to jszip that don't exist. 

Whenever I want to test saving the standalone gallery I usually do a `build`, then update the `save-gallery.vue` → `files` array using the exact files from the `dist/gallery/chunks` folder and build again. But like I said, I only do this if I'm specifically testing the standalone gallery saving or something in the standalone gallery, so there are likely a kazillion commits where the saving process just doesn't work in the `dist-zip`.

The other thing is that if you're serving the files using `watch:dev`, it gives you a slightly different set of chunk files and so obviously the `save-gallery.vue` → `files` array is not going to match at all. I've taken to only testing the standalone gallery in a complete `build` to give it a fair chance. So I'd recommend that.

I'm fairly confident the issue lies somewhere in there given the changes you made, and on that note:

`.[` is definitely a syntax error, but I was surprised it had not caused any issues so far in my tests. So I looked at my test gallery [here](https://joonaspaakko.github.io/ale-test-new/gallery.1632088147952.js) (you can prettify the code in Chrome's dev tools sources tab... and search for `buildStandaloneData:`) and I noticed that it's actually fixed there in my built code... perhaps when the code is minified? I'm not sure, but what I do know is that when I use the commands `watch:dev` or `build`, somewhere during these processes the issue gets fixed (in my end anyways).

> The thing I can't figure out is why this would be any different on your end...

![image](https://user-images.githubusercontent.com/1164476/135767717-371c5a86-c047-4e12-a10a-15a295a1f15b.png)

But then there's the second thing, which is the standalone gallery saving process. Whether or not this `.[`is an actual issue in the built code, the thing is, it's not related to the saving process. It's related to opening the gallery pages in the standalone gallery.

And if I'm reading the changes that you made in the `save-gallery.vue` file correctly, it's suppressing errors... and while that might help you get through the saving process, I'm not overly confident that it would result in fully functional standalone gallery.

-----

**So in conclusion:**

1. I think the issue has to do with quirks in the dev environment.
2. The `.[` syntax errors, which I found a few more of don't seem to be an issue, but there's still no reason not to fix them.
3. I don't think suppressing jszip errors is a good approach or necessary.

P.S. I think your editor might have the `trimTrailingWhitespace` setting on, which is causing the  whitespace changes. I don't like how that setting doesn't just remove the whitespace at the end of a line, but it also removes indentation on empty lines. When the code comes out minified anyways, I don't much care how much whitespace exists in the source files so I keep it disabled. I will be adding `.vscode/settings.json` → `"files.trimTrailingWhitespace": false`.

Reply:
You're right, I was testing with `watch:dev`!  That makes sense.  

In my testing, I found that you could pass a function to `chunkFilename`; I bet we could use that.  `[name]` falls back to `[id]`, but maybe we could be like "Well, if you don't have a name, use this known value instead".

With respect to the syntax error, I just happened to see it highlighted in my editor and was surprised that webpack didn't blow up in the first place.

Speaking of blowing up in the first place, you're right about blowing up instead of silently failing / doing part of the job.

With respect to editors, I use IntelliJ IDEA for javascript (and actually use Rider for .net stuff now too).